spring-integration-core/src/main/java/org/springframework/integration/dispatcher/AbstractDispatcher.java

```diff
@@ -123,6 +123,23 @@ protected boolean tryOptimizedDispatch(Message<?> message) {
 		return false;
 	}
 
+	/**
+	 * If the exception is not a {@link MessagingException} or does not have a
+	 * {@link MessagingException#getFailedMessage() failedMessage}, wrap it in a new
+	 * {@link MessagingException} with the message. There is some inconsistency here in
+	 * that {@link MessagingException}s are wrapped in a {@link MessagingException} whereas
+	 * {@link Exception}s are wrapped in {@link MessageDeliveryException}. It is retained
+	 * for backwards compatibility and will be resolved in 5.1.
+	 * It also does not wrap other {@link RuntimeException}s.
+	 * TODO: Remove this in favor of
+	 * {@code #wrapInDeliveryExceptionIfNecessary(Message, Supplier, Exception)} in 5.1.
+	 * @deprecated in favor of
```

Comment:
I think `@deprecated` must be in the end.
See Checkstyle rule:
```
<!--  		<module name="AtclauseOrder"> -->
<!-- 			<property name="target" value="METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/> -->
<!--     		<property name="tagOrder" value="@param, @return, @throws, @since, @deprecated, @see"/> -->
<!-- 		</module> -->
```

Right, it is commented out, but may be we should consider to switch it on already...